### PR TITLE
Add thread radio coex metrics support

### PR DESCRIPTION
**Pull request by zhanglongxia**
This commit adds  "SPINEL_PROP_RADIO_COEX_METRICS" to the list of spinel properties. It adds "kWPANTUNDProperty_NCPCoexMetrics" and "kWPANTUNDProperty_NCPCoexMetricsAsValMap" wpan properties to get the radio coex metrics.

### CODE REVIEW

File: third_party/openthread/src/ncp/spinel.h [hunk 1]
```diff
@@ -1627,6 +1627,43 @@ typedef enum
      */
     SPINEL_PROP_RADIO_CAPS = SPINEL_PROP_PHY_EXT__BEGIN + 11,
 
+    /// All coex metrics related counters.
+    /** Format: Lbt(LLLLLLLL)t(LLLLLLLLL)  (Read-only)
```
Comment (bukepo) [inline third_party/openthread/src/ncp/spinel.h]: `Lb` or `bL`?

File: src/ncp-spinel/SpinelNCPInstance.cpp [hunk 2]
```diff
@@ -1042,6 +1043,123 @@ unpack_ncp_counters_mle(const uint8_t *data_in, spinel_size_t data_len, boost::a
...
+			);
+
+			require_action(len > 0, bail, ret = kWPANTUNDStatus_Failure);
+
+			struct_in  += len;
+			struct_len -= len;
+
+			if (!as_val_map) {
+				char c_string[200];
+				snprintf(c_string, sizeof(c_string), "%-20s = %d", *counter_names, counter_value);
```
Comment (bukepo) [inline src/ncp-spinel/SpinelNCPInstance.cpp]: `d` -> `u`.

File: src/ncp-spinel/SpinelNCPInstance.cpp [hunk 3]
```diff
@@ -1042,6 +1043,123 @@ unpack_ncp_counters_mle(const uint8_t *data_in, spinel_size_t data_len, boost::a
...
 
+static int
+unpack_coex_metrics(const uint8_t *data_in, spinel_size_t data_len, boost::any& value, bool as_val_map)
+{
+	std::list<std::string> result_as_string;
+	ValueMap result_as_val_map;
+	int ret = kWPANTUNDStatus_Ok;
+	spinel_ssize_t len;
+	bool stopped;
+	uint32_t numGrantGlitch;
```
Comment (bukepo) [inline src/ncp-spinel/SpinelNCPInstance.cpp]: not sure if we should unify to use snake case.

File: third_party/openthread/src/ncp/spinel.h (hunk 1 above)
Comment (zhanglongxia) [reply third_party/openthread/src/ncp/spinel.h]: It should be `bL`. I will update it.

File: src/ncp-spinel/SpinelNCPInstance.cpp (hunk 2 above)
Comment (zhanglongxia) [reply src/ncp-spinel/SpinelNCPInstance.cpp]: Will update it.

File: src/ncp-spinel/SpinelNCPInstance.cpp (hunk 3 above)
Comment (zhanglongxia) [reply src/ncp-spinel/SpinelNCPInstance.cpp]: I have updated the name to keep the code style the same.

**Comment by zhanglongxia** [comment]
@abtink Could you help me review this PR at your convenience?

**Comment by abtink** [comment]
Related PR in OpenThread: https://github.com/openthread/openthread/pull/4108

**Pull request merged**